lib/browser/fingerprint.ts

```diff
@@ -17,31 +17,38 @@ import {
   addListener,
   removeListener
 } from '../oidc';
-import { FingerprintOptions } from '../authn/types';
+import { FingerprintOptions } from '../base/types';
 import { OktaAuthHttpInterface } from '../http/types';
 
-export default function fingerprint(sdk: OktaAuthHttpInterface, options?: FingerprintOptions): Promise<string> {
-  options = options || {};
+const isMessageFromCorrectSource = (iframe: HTMLIFrameElement, event: MessageEvent)
+: boolean => event.source === iframe.contentWindow;
 
+export default function fingerprint(sdk: OktaAuthHttpInterface, options?: FingerprintOptions): Promise<string> {
   if (!isFingerprintSupported()) {
     return Promise.reject(new AuthSdkError('Fingerprinting is not supported on this device'));
   }
 
-  var timeout;
-  var iframe;
-  var listener;
-  var promise = new Promise(function (resolve, reject) {
+  const container = options?.container ?? document.body;
+  let timeout: NodeJS.Timeout;
+  let iframe: HTMLIFrameElement;
+  let listener: (this: Window, ev: MessageEvent) => void;
+  let msg;
```

Comment:
is `msg` used outside of `listener`? If not, we should move into the `listener` function